solr/core/src/test-files/solr/solr-50-all.xml

```diff
@@ -76,4 +76,9 @@
     <repository name="local" class="a.b.C" default="true"/>
   </backup>
 
+  <clusterPluginsSource class="org.apache.solr.api.NodeConfigClusterPluginsSource"/>
+
+  <clusterSingleton name="testSingleton1" class="my.singleton.class"/>
+  <clusterSingleton name="testSingleton2" class="my.other.singleton.class"/>
```

Comment:
I changed the element name here to `clusterSingleton`, but at present there is nothing that actually enforces the constraint that the declared class implements the `ClusterSingleton` interface. Theoretically, this would instantiate and manage the provided class in a manner that is consistent with it's class hierarchy, and not the `clusterSingleton` tag given here.

If we consider the equivalent operation of adding a plugin using the /cluster/plugin API, there is no part of the request that identifies the plugin as a cluster singleton.

If we go with this naming, should there also be a check somewhere that the class actually implements `ClusterSingleton`?

Reply:
Enforcement would be ideal but not essential.